src/prefect/server/api/task_runs.py

```diff
@@ -279,19 +283,39 @@ class TaskQueue:
     _task_queues: Dict[str, Self] = {}
     _scheduled_tasks_already_restored: bool = False
 
-    task_key: str
+    default_scheduled_max_size: int = (
+        PREFECT_TASK_SCHEDULING_MAX_SCHEDULED_QUEUE_SIZE.value()
+    )
+    default_retry_max_size: int = PREFECT_TASK_SCHEDULING_MAX_RETRY_QUEUE_SIZE.value()
 
+    _queue_size_configs: Dict[str, Tuple[int, int]] = {}
+
+    task_key: str
     _scheduled_queue: asyncio.Queue
     _retry_queue: asyncio.Queue
 
     @classmethod
     async def enqueue(cls, task_run: schemas.core.TaskRun) -> None:
         await cls.for_key(task_run.task_key).put(task_run)
 
+    @classmethod
+    def configure_task_key(
```

Comment:
Ooo I like that you made it configurable per task-key, that's a great steam valve for users.